### DISCUSSION TITLE
mac os (intel x86) build lotus generates warning "could not create compact unwind for _blst_sha256_block_data_order : does not use RBP or RSP based frame"

### DISCUSSION THREAD

**Discussion by Icarus9913** [filecoin-project/lotus#7676] [General Q&A] [answered]
cpu: intel i5

env:
> export IPFS_GATEWAY=https://proof-parameters.s3.cn-south-1.jdcloud-oss.com/ipfs/
> export GOPROXY=https://goproxy.cn
> export CGO_CFLAGS_ALLOW="-D__BLST_PORTABLE__"
> export CGO_CFLAGS="-D__BLST_PORTABLE__"

dependencies: `brew install go bzr jq pkg-config rustup hwloc`

command: `make clean && make all`

logs:

> \+ eval '__tarball_path='\''/tmp/filecoin-ffi-Darwin-standard_47854851.tar.gz'\'''
> \++ __tarball_path=/tmp/filecoin-ffi-Darwin-standard_47854851.tar.gz
> \++ mktemp -d
> \+ local __tmp_dir=/var/folders/1q/7gf23bkx2ps2gjzwvwldjylw0000gn/T/tmp.56fBOuRT
> \+ tar -C /var/folders/1q/7gf23bkx2ps2gjzwvwldjylw0000gn/T/tmp.56fBOuRT -xzf /tmp/filecoin-ffi-Darwin-standard_47854851.tar.gz
> \+ find -L /var/folders/1q/7gf23bkx2ps2gjzwvwldjylw0000gn/T/tmp.56fBOuRT -type f -name filcrypto.h -exec cp -- '{}' . ';'
> \+ find -L /var/folders/1q/7gf23bkx2ps2gjzwvwldjylw0000gn/T/tmp.56fBOuRT -type f -name libfilcrypto.a -exec cp -- '{}' . ';'
> \+ find -L /var/folders/1q/7gf23bkx2ps2gjzwvwldjylw0000gn/T/tmp.56fBOuRT -type f -name filcrypto.pc -exec cp -- '{}' . ';'
> \+ check_installed_files
> \+ pwd
> \/Users/icaruswu/gocode/src/lotus/extern/filecoin-ffi
> \+ ls ./filcrypto.h ./filcrypto.pc ./filcrypto.yml ./install-filcrypto ./libfilcrypto.a
> \./filcrypto.h       ./filcrypto.pc      ./filcrypto.yml     ./install-filcrypto ./libfilcrypto.a
> \+ [[ ! -f ./filcrypto.h ]]
> \+ [[ ! -f ./libfilcrypto.a ]]
> \+ [[ ! -f ./filcrypto.pc ]]
> \+ echo '[install-filcrypto/main] successfully installed prebuilt libfilcrypto'
> \[install-filcrypto/main] successfully installed prebuilt libfilcrypto
> \+ exit
rm -f lotus
go build  -ldflags="-X=github.com/filecoin-project/lotus/build.CurrentCommit=+git.37141779a.dirty" -o lotus ./cmd/lotus
> \# github.com/filecoin-project/filecoin-ffi/generated
ld: warning: could not create compact unwind for _blst_sha256_block_data_order: does not use RBP or RSP based frame
> \# github.com/filecoin-project/lotus/cmd/lotus
ld: warning: could not create compact unwind for _blst_sha256_block_data_order: does not use RBP or RSP based frame
rm -f lotus-miner
go build  -ldflags="-X=github.com/filecoin-project/lotus/build.CurrentCommit=+git.37141779a.dirty" -o lotus-miner ./cmd/lotus-miner
> \# github.com/filecoin-project/lotus/cmd/lotus-miner
ld: warning: could not create compact unwind for _blst_sha256_block_data_order: does not use RBP or RSP based frame
rm -f lotus-worker
go build  -ldflags="-X=github.com/filecoin-project/lotus/build.CurrentCommit=+git.37141779a.dirty" -o lotus-worker > \./cmd/lotus-seal-worker
> \# github.com/filecoin-project/lotus/cmd/lotus-seal-worker
ld: warning: could not create compact unwind for _blst_sha256_block_data_order: does not use RBP or RSP based frame

**Comment by PerrywLiu** [comment, marked as answer]
it can works normal

**Comment by Icarus9913** [comment]
If I ignore them, that will make me can not debug the codes.
Here's the logs.

> \# pkg-config --cflags  -- /Users/icaruswu/gocode/pkg/mod/github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f/generated/../filcrypto.pc /Users/icaruswu/gocode/pkg/mod/github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f/generated/../filcrypto.pc /Users/icaruswu/gocode/pkg/mod/github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f/generated/../filcrypto.pc /Users/icaruswu/gocode/pkg/mod/github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f/generated/../filcrypto.pc > \/Users/icaruswu/gocode/pkg/mod/github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f/generated/../filcrypto.pc
> Failed to open '/Users/icaruswu/gocode/pkg/mod/github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f/generated/../filcrypto.pc': No such file or directory
> No package '/Users/icaruswu/gocode/pkg/mod/github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f/generated/../filcrypto.pc' found
> Failed to open '/Users/icaruswu/gocode/pkg/mod/github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f/generated/../filcrypto.pc': No such file or directory
> No package '/Users/icaruswu/gocode/pkg/mod/github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f/generated/../filcrypto.pc' found
> Failed to open '/Users/icaruswu/gocode/pkg/mod/github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f/generated/../filcrypto.pc': No such file or directory
> No package '/Users/icaruswu/gocode/pkg/mod/github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f/generated/../filcrypto.pc' found
> Failed to open '/Users/icaruswu/gocode/pkg/mod/github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f/generated/../filcrypto.pc': No such file or directory
> No package '/Users/icaruswu/gocode/pkg/mod/github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f/generated/../filcrypto.pc' found
> Failed to open '/Users/icaruswu/gocode/pkg/mod/github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f/generated/../filcrypto.pc': No such file or directory
> No package '/Users/icaruswu/gocode/pkg/mod/github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f/generated/../filcrypto.pc' found
> pkg-config: exit status 1

**Reply by LexLuthr** [reply]
Are you running this on a M1 chip or older Intel based Mac?

**Reply by Icarus9913** [reply]
Intel(R) Core(TM) i5-1038NG7 CPU @ 2.00GHz
X86

**Reply by LexLuthr** [reply]
What is the operating system version?

**Reply by Icarus9913** [reply]
macOS Big Sur 11.3.1.
Macbook Pro 2020. 
There are some people met this issue too. But da-you(WeiChen) said that's ok with his machine. 
And please check my second logs cause I can't debug the codes to connect with lotus daemon.

**Reply by LexLuthr** [reply]
We can try to compile FFI manually to check if that resolves the linker warning.

export LIBRARY_PATH=/opt/homebrew/lib
export FFI_BUILD_FROM_SOURCE=1

make clean && make all
